crates/polars-plan/src/logical_plan/aexpr/schema.rs

```diff
@@ -127,7 +127,11 @@ impl AExpr {
                     Mean(expr) => {
                         let mut field =
                             arena.get(*expr).to_field(schema, Context::Default, arena)?;
-                        float_type(&mut field);
+                        if matches!(&field.dtype, DataType::Boolean) {
```

Comment:
This branch already happens in the `float_type` call?

Reply:
I'll be honest, I just blindly copied @advoet's code from the linked issue. But it does fix the bug.

Reply:
It does fix it, but it is a bandaid.

Reply:
Happy to implement a more robust fix, just let me know where the "right" place to change would be. It's an important fix because people like to use 100 * boolean_column.mean() to calculate percentages.

Reply:
There are a _lot_ of cases where boolean is treated as a `u8` in these types of operations. It might be worth adding `is_numeric_or_bool()` for those many cases scattered throughout the codebase (perhaps different PR).

Reply:
Eh, I just perused through and it feels like it's not worth it. Go ahead with this one.

Reply:
ohhh I didn't see Ritchie's edit to his original post! thanks :)

Reply:
He may have been noting the [other `float_type` function in `expr_dyn_fn.rs`](https://github.com/pola-rs/polars/blob/main/crates/polars-plan/src/dsl/expr_dyn_fn.rs) but this is actually never used.